File: app/src/main/java/com/example/pshiksha/transactions/OrderDetails.java
package com.example.pshiksha.transactions;

import java.io.Serializable;

public class OrderDetails implements Serializable {
    private final String userUid;
    private final String orderId;
    private final String serviceName;
    private final ServiceTransaction serviceTransaction;

    public OrderDetails() {
        userUid = "INVALID ID";
        orderId = "INVALID ID";
        serviceName = "INVALID NAME";
        serviceTransaction = null;
    }

    public OrderDetails(String userUid, String orderId, String serviceName, ServiceTransaction serviceTransaction) {
        this.userUid = userUid;
        this.orderId = orderId;
        this.serviceName = serviceName;
        this.serviceTransaction = serviceTransaction;
    }

    public String getUserUid() {
        return userUid;
    }

    public String getOrderId() {
        return orderId;
    }

    public String getServiceName() {
        return serviceName;
    }

    public ServiceTransaction getServiceTransaction() {
        return serviceTransaction;
    }
}

File: app/src/main/java/com/example/pshiksha/main/orders/OrderDetails.kt
package com.example.pshiksha.main.orders

import java.io.Serializable

class OrderDetails : Serializable {
    var userUid: String
    var orderId: String
    var serviceName: String
    var serviceTransaction: TransactionDetails
    var serviceDescription: String
    var serviceStatus: ServiceStatus

    constructor() {
        userUid = "NOT AVAILABLE"
        orderId = "NOT AVAILABLE"
        serviceName = "NOT AVAILABLE"
        serviceTransaction = TransactionDetails()
        serviceDescription = "NOT AVAILABLE"
        serviceStatus = ServiceStatus.PENDING
    }

    constructor(
        userUid: String,
        orderId: String,
        serviceName: String,
        transactionDetails: TransactionDetails,
        serviceDescription: String,
        serviceStatus: ServiceStatus
    ) {
        this.userUid = userUid
        this.orderId = orderId
        this.serviceName = serviceName
        serviceTransaction = transactionDetails
        this.serviceDescription = serviceDescription
        this.serviceStatus = serviceStatus
    }

    enum class ServiceStatus {
        PENDING, COMPLETED
    }
}